### PR TITLE
Created an example of using external security

### PR DESCRIPTION
**Pull request by freshie**
I created an issue for this, issue #347. (https://github.com/marklogic-community/ml-gradle/issues/347)

### CODE REVIEW

File: examples/external-security/README.md [hunk 1]
```diff
@@ -0,0 +1,4 @@
+This project shows how to configure an external security and use it in an app sever. 
+
+If you want to deploy using a user with external security you'll need to configure the 8000, 8001 and 8002 server added of time.
```
Comment (rjrudin) [inline examples/external-security/README.md]: "added" should be "ahead"?

File: examples/external-security/build.gradle [hunk 2]
```diff
@@ -0,0 +1,13 @@
+plugins {
+	// The properties plugin must be applied BEFORE ml-gradle so that ml-gradle sees the results of the properties plugin
```
Comment (rjrudin) [inline examples/external-security/build.gradle]: I think this is documented elsewhere and can be removed - I like to slim these example projects down to the bare minimum so it's clear what's being demonstrated.

File: examples/external-security/src/main/ml-config/servers/example.json [hunk 3]
```diff
@@ -0,0 +1,16 @@
+{
```
Comment (rjrudin) [inline examples/external-security/src/main/ml-config/servers/example.json]: Can you format this file? 2 spaces or a tab, doesn't really matter.

File: examples/external-security/README.md [hunk 4]
```diff
@@ -0,0 +1,4 @@
+This project shows how to configure an external security and use it in an app sever.
```
Comment (rjrudin) [inline examples/external-security/README.md]: "sever" -> "server"

File: examples/external-security/build.gradle [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+	// processing the gradle-(environmentName).properties files.
+	id "net.saliman.properties" version "1.4.6"
+
+	id "com.marklogic.ml-gradle" version "3.6.0"
+}
+
+task printHost {
```
Comment (rjrudin) [inline examples/external-security/build.gradle]: Is this task needed for the purpose of showing how to configure an external security?

File: examples/external-security/gradle-dev.properties [hunk 6]
```diff
@@ -0,0 +1,8 @@
+mlHost=dev-host
```
Comment (rjrudin) [inline examples/external-security/gradle-dev.properties]: I don't think it's necessary to show different environment properties files in action - I think it distracts from the purpose of showing how to configure external security. I think just use "gradle.properties", no need for the different environment files.

File: examples/external-security/src/main/ml-config/servers/example.json [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+	"server-type": "http",
+  	"root": "/",
+  	"group-name": "%%GROUP%%",
+  	"port": "5432",
+	"server-name": "example-servers",
+	"modules-database": "%%mlModulesDbName%%",
```
Comment (rjrudin) [inline examples/external-security/src/main/ml-config/servers/example.json]: Is "mlModulesDbName" defined somewhere?

File: examples/external-security/gradle.properties [hunk 8]
```diff
@@ -0,0 +1,9 @@
+mlHost=localhost
+mlAppName=properties-example
```
Comment (rjrudin) [inline examples/external-security/gradle.properties]: How about "external-security-example"?

**Comment by freshie** [comment]
Made requested changes.

**Comment by rjrudin** [comment]
Thanks @freshie I will try it out this week.

**Pull request merged**